x-pack/plugins/kubernetes_security/public/components/count_widget/index.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import { EuiFlexGroup, EuiFlexItem, EuiText, EuiLoadingSpinner, EuiIconTip } from '@elastic/eui';
+import { CountResult } from '../../../common/types/count';
+import { useStyles } from './styles';
+import type { IndexPattern, GlobalFilter } from '../../types';
+import { addTimerangeToQuery } from '../../utils/add_timerange_to_query';
+import { useFetchCountWidgetData } from './hooks';
+import { addResourceTypeToFilterQuery, numberFormatter } from './helpers';
+
+export const LOADING_TEST_ID = 'kubernetesSecurity:count-widget-loading';
+export const TOOLTIP_TEST_ID = 'kubernetesSecurity:count-widget-tooltip';
+
+export interface CountWidgetDeps {
+  title: string;
+  widgetKey: string;
+  indexPattern?: IndexPattern;
+  globalFilter: GlobalFilter;
+  groupedBy: string;
+}
+
+export const CountWidget = ({
+  title,
+  widgetKey,
+  indexPattern,
+  globalFilter,
+  groupedBy,
+}: CountWidgetDeps) => {
+  const styles = useStyles();
+
+  const filterQueryWithTimeRange = useMemo(() => {
+    let globalFilterModified = globalFilter.filterQuery;
+
+    if (widgetKey === 'CountNodesWidgets') {
+      globalFilterModified = addResourceTypeToFilterQuery(globalFilter.filterQuery, 'node');
+    }
+
+    if (widgetKey === 'CountPodsWidgets') {
+      globalFilterModified = addResourceTypeToFilterQuery(globalFilter.filterQuery, 'pod');
+    }
+    return addTimerangeToQuery(globalFilterModified, globalFilter.startDate, globalFilter.endDate);
+  }, [globalFilter.filterQuery, globalFilter.startDate, globalFilter.endDate, widgetKey]);
+
+  const { data, isLoading } = useFetchCountWidgetData(
+    widgetKey,
+    filterQueryWithTimeRange,
+    groupedBy,
+    indexPattern?.title
+  );
+
+  const countValue = useMemo((): CountResult => {
+    return data ? data?.pages[0] : (0 as unknown as CountResult);
```

Comment:
what is `data.pages[0]`? Is it `CountResult`? why not just get the value of `CountResult` which is a `number` so that it's easier to work with (also fits into the name `countValue` better), and you won't need to cast numbers to `unknown` and then `CountResult` anywhere

Reply:
If `countValue` is a `CountResult` which is an object under the hood, passing it as `content` into `EuiTooltip` will run into errors on hover
<img width="116" alt="image" src="https://user-images.githubusercontent.com/16872649/178364568-e98e67a6-9d09-4c76-8098-fd83f49ebf25.png">

<img width="605" alt="image" src="https://user-images.githubusercontent.com/16872649/178364415-5e7a31bf-af3d-4954-aaf1-5e063d52740f.png">